NVIDIA/deepops
# GPU Operator 1.9 support with DGX inclusion

* Bump GPU Operator 
*  Bump Device Plugin 
*  Remove node-prep role as GPU Operator now handles that logic internally
*  Introduce flags to enable/disable gpu operator components (migmanager, dcgm, driver, toolkit)
* Dynamically detect DGX systems and disable toolkit & driver features

Test matrix:
* Install K8s on DGX, verify the system is properly detected and the GPU jobs run
*  Install on non-DGX without GPU Operator, verify new device plugin versions work
*  Install on non-DGX with GPU Operator, verify new gpu operator versions work
*  Install with components disabled to verify flags are pushed through

TODO:

- [x] GPU Operator monitoring integration needs to be completed before the gpu operator install can be set as the default
- [x] Additional tests should be introduced to uninstall and re-install the gpu operator with different flags
- [x] The automated testing should be double checked to make sure we aren't double testing things
- [x] NVIDIA AI Enterprise does not yet support GPU Operator v1.9.0 and we should hold of on merging this for a few weeks until that support is added.
